tests/validation/tests/v1_api/test_monitoring_v2.py

```diff
@@ -0,0 +1,94 @@
+from .common import *  # NOQA
+import pytest
+
+namespace = {'client': None, 'cluster': None, 'project': None}
+monitoring_v2_version = os.environ.get('RANCHER_MONITORING_V2_VERSION', '9.4.202')
+monitoring_v2_app_id = 'cattle-monitoring-system/rancher-monitoring'
+monitoring_v2_crd_app_id = 'cattle-monitoring-system/rancher-monitoring-crd'
+monitoring_v2_namespace = "cattle-monitoring-system"
+
+
+def test_install_monitoring_v2():
+    install_monitoring()
+
+
+def install_monitoring():
+    client = namespace['client']
+    a = client.list_catalog_cattle_io_clusterrepo()
+    rancher_catalog = client.by_id_catalog_cattle_io_clusterrepo(id="rancher-charts")
+    if rancher_catalog is None:
+        assert False, "rancher-charts is not available"
+
+    # install the monitoring v2 app into the new project we just created
+    project_id = namespace["project"]["id"]
+    cluster_id = namespace["cluster"]["id"]
+    cluster_name = namespace["cluster"]["spec"]["displayName"]
+    values = read_json_from_resource_dir("monitoring_v2_values.json")
+    values["projectId"] = project_id
+    for chart in values["charts"]:
+        chart["version"] = monitoring_v2_version
+        chart["projectId"] = project_id
+        chart["values"]["global"]["cattle"]["clusterId"] = cluster_id
+        chart["values"]["global"]["cattle"]["clusterName"] = cluster_name
+    res = client.action(rancher_catalog, "install", values)
+    display(res)
+    # wait 2 minutes for the app to be fully deployed
+    time.sleep(120)
+    # check the app rancher-monitoring-crd first, then the app rancher-monitoring
+    wait_for(lambda: client.by_id_catalog_cattle_io_app(monitoring_v2_crd_app_id).status.summary.state == "deployed",
+             timeout_message="time out waiting for app to be ready")
+    wait_for(lambda: client.by_id_catalog_cattle_io_app(monitoring_v2_app_id).status.summary.state == "deployed",
+             timeout_message="time out waiting for app to be ready")
+
+
+def uninstall_monitoring():
+    client = namespace['client']
+    # firstly, uninstall the monitoring app
+    app = client.by_id_catalog_cattle_io_app(monitoring_v2_app_id)
+    if app is not None:
+        client.action(app, "uninstall")
+        wait_for(lambda: client.by_id_catalog_cattle_io_app(monitoring_v2_app_id) is None)
+    # then, clean up the secrets left in the namespace
```

Comment:
Just wondering if this step is also done when we uninstall from Dashboard.

Reply:
There is no button for us to uninstall monitoring with one click. 
We need to uninstall the monitoring first, then the monitroing_crd from Dashboard

Reply:
Yes this happens when will click "Delete" from the Dashboard on the app

Reply:
Sorry I wasn't clear I was referring to the Secrets management that happens in between CRD/Chart uninstalls.

Reply:
@izaac  If the secrets are not deleted, we will not be able to delete the namespace: the namespace will be stuck in the deleting state.

Reply:
Ok let's make sure we delete the namespace. I'm not sure if there are other charts that leave the namespace after an uninstall. 
I was curious in this specific case of monitoring if the chart deletes the namespace why are we deleting it explicitly in this test.

Reply:
oh good point. I have an [issue](https://github.com/rancher/charts/issues/1038) logged by our team(yet to follow up) which states deleting monitoring v2 leaves behind namespace

Reply:
> Ok let's make sure we delete the namespace. I'm not sure if there are other charts that leave the namespace after an uninstall.

This is case by case. For monitoring, it creates 2 namespaces: cattle-monitoring-system and cattle-dashboards, but leave the cattle-monitoring-system after uninstalling it.